### PR TITLE
feat: separate future and expired checks

### PR DESCRIPTION
**Pull request by SalladinBalwer**
As the RFC says that proofs signed in the future MAY be accepted, the default behavior should then be that a proof signed in the future should NOT be accepted. [See section 11.1-4](https://datatracker.ietf.org/doc/html/rfc9449#section-11.1-4)

This PR adds a new default value for allowed proof age, `DEFAULT_ALLOWED_PROOF_AGE`, that is set to a more generous 5 minutes. It also changes the existing default time window into the future to 0 seconds.

Now if a `TimeWindow` is not set in the `ParseOptions`, future proofs will all be rejected. The functionality to control how long a proof is allowed to live is controlled by the new parse option `AllowedProofAge`.

### Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have verified that the code builds perfectly fine on my local system
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have verified that my code follows the style already available in the repository
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by SalladinBalwer** [comment]
I appreciate the effort @chrlmrtnssn but I need an approval from a fellow maintainer. 

@a354dpa @EmmaAndersen Can you help me out?

**Review by a354dpa — APPROVED** [review]

**Pull request merged**
